crates/amaru-network/src/network_resource.rs

```diff
@@ -12,101 +12,226 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::acto_connection;
+use crate::server::downstream_server::DownstreamServer;
+use crate::upstream_connection;
 use acto::{AcTokioRuntime, ActoRef, ActoRuntime};
+use amaru_kernel::is_header::HeaderTip;
 use amaru_kernel::{
     BlockHeader, Point,
-    connection::{BlockFetchClientError, ConnMsg},
+    connection::{ClientConnectionError, ConnMsg},
     consensus_events::{ChainSyncEvent, Tracked},
     peer::Peer,
 };
 use amaru_ouroboros::ChainStore;
-use amaru_ouroboros_traits::NetworkOperations;
+use amaru_ouroboros_traits::network_operations::ForwardEvent;
+use amaru_ouroboros_traits::{NetworkOperations, TxClientReply, TxServerRequest};
+use anyhow::anyhow;
 use async_trait::async_trait;
 use parking_lot::Mutex;
+use std::time::Duration;
 use std::{collections::BTreeMap, ops::Deref, sync::Arc};
 use tokio::sync::{mpsc, oneshot};
+use tracing::warn;
 
 #[derive(Clone)]
 pub struct NetworkResource {
-    inner: Arc<NetworkInner>,
+    upstream_client: Arc<UpstreamClient>,
+    downstream_server: Arc<DownstreamServer>,
 }
 
 impl Deref for NetworkResource {
-    type Target = NetworkInner;
+    type Target = UpstreamClient;
```

Comment:
in general, this Deref should go because there are two viable targets, but I see this as a simplification of working on this while the real network stack isn’t there yet, right?